core/services/gateway/handlers/functions/orm.go

```diff
@@ -130,3 +136,46 @@ func (cs *cachedSubscriptionRow) encode() CachedSubscription {
 		},
 	}
 }
+
+func (o *orm) GetAllowedSenders(offset, limit uint, qopts ...pg.QOpt) ([]common.Address, error) {
+	var addresses []common.Address
+	stmt := fmt.Sprintf(`
+		SELECT allowed_address
+		FROM %s
+		WHERE router_contract_address = $1
+		ORDER BY id ASC
+		OFFSET $2
+		LIMIT $3;
+	`, allowlistTableName)
+	err := o.q.WithOpts(qopts...).Select(&addresses, stmt, o.routerContractAddress, offset, limit)
+	if err != nil {
+		return addresses, err
+	}
+	o.lggr.Debugf("Successfully fetched allowed sender list from DB. offset: %d, limit: %d, length: %d", offset, limit, len(addresses))
+
+	return addresses, nil
+}
+
+func (o *orm) CreateAllowedSender(allowedSender []common.Address, qopts ...pg.QOpt) error {
+	var valuesPlaceholder []string
+	for i := 1; i <= len(allowedSender)*2; i += 2 {
+		valuesPlaceholder = append(valuesPlaceholder, fmt.Sprintf("($%d, $%d)", i, i+1))
+	}
+
+	stmt := fmt.Sprintf(`
+		INSERT INTO %s (allowed_address, router_contract_address)
+		VALUES %s ON CONFLICT (allowed_address, router_contract_address) DO NOTHING;`, allowlistTableName, strings.Join(valuesPlaceholder, ", "))
```

Comment:
How are allowlist removals handled?

Reply:
From what I saw this only happens in the contract whenever the `blockSender` method is called. This removes the allowed_address from the allowlist and adds it to the `blockedSenders` one. This method does not seem to be used in our go code, despite having the wrapper for it. is this happening externally? if thats the case, it will be hard to detect the deletions only by the absence of it while querying a batch.

Reply:
reviewing the code again what we are currently doing is fetching every allowed address and replacing the whole list whenever we do: `a.update(allowedSenderList)` this is why the values are up to date. If we want to keep this behaviour probably I should modify the orm layer to replicate the same on the DB with a `UpdateAllowedSenders` instead of `CreateAllowedSenders`.